server/bleep/src/remotes.rs

```diff
@@ -75,6 +96,79 @@ impl From<git2::Error> for RemoteError {
     }
 }
 
+#[derive(Serialize, Deserialize, ToSchema, PartialEq, Eq, Clone, Debug)]
+#[serde(rename_all = "snake_case")]
+pub enum RepoRemote {
+    Git(GitRemote),
+    None,
+}
+
+impl<T: AsRef<RepoRef>> From<T> for RepoRemote {
+    fn from(reporef: T) -> Self {
+        match reporef.as_ref() {
+            RepoRef {
+                backend: Backend::Github,
+                name,
+            } => RepoRemote::Git(GitRemote {
+                protocol: GitProtocol::Https,
+                host: "github.com".to_owned(),
+                address: name.to_owned(),
+            }),
+            RepoRef {
+                backend: Backend::Local,
+                name: _name,
+            } => RepoRemote::None,
+        }
+    }
+}
+
+impl Display for RepoRemote {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            RepoRemote::Git(GitRemote {
+                protocol,
+                host,
+                address,
+            }) => match protocol {
+                GitProtocol::Https => write!(f, "https://{host}/{address}.git"),
+                GitProtocol::Ssh => write!(f, "git@{host}:{address}.git"),
+            },
+            RepoRemote::None => write!(f, "none"),
+        }
+    }
+}
+
+impl FromStr for RepoRemote {
+    type Err = ();
+
+    fn from_str(value: &str) -> std::result::Result<Self, Self::Err> {
+        if let Some(stripped) = value.strip_prefix("https://github.com/") {
+            return Ok(RepoRemote::Git(GitRemote {
+                protocol: GitProtocol::Https,
+                host: "github.com".to_owned(),
+                address: stripped
+                    .trim_end_matches('/')
+                    .trim_end_matches(".git")
+                    .to_owned(),
+            }));
+        }
+
+        if let Some(stripped) = value.strip_prefix("git@github.com:") {
+            return Ok(RepoRemote::Git(GitRemote {
+                protocol: GitProtocol::Ssh,
+                host: "github.com".to_owned(),
+                address: stripped
+                    .trim_start_matches('/')
+                    .trim_end_matches('/')
+                    .trim_end_matches(".git")
+                    .to_owned(),
+            }));
+        }
+
+        Err(())
+    }
+}
+
```

Comment:
This doesn't logically belong here. This is merely a descriptor of the repository's properties and doesn't have functional part in remote interaction. I'd prefer if these travelled with `Repository`. If you think it's too noisy in one module, submodules would work well.